### PR TITLE
Random advanced distributions

### PR DESCRIPTION
**Pull request by fduguet-nv**

### CODE REVIEW

**Comment by manopapad** [comment]
@fduguet-nv Please merge from branch-22.07 to get mypy checking (and add types to your python functions).

**Review by magnatelee — APPROVED** [review]

**Comment by magnatelee** [comment]
@fduguet-nv feel free to merge this PR once you update the code to pass the mypy checking as @manopapad requested.

Once this PR is merged, I'll do a follow-on refactoring on the bitgenerator code and probably update the legacy APIs to make them use the new bitgenerator code.

**Pull request merged**
